sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/samples/TimeSeriesInsightsClientSample/InstancesSamples.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using static Azure.IoT.TimeSeriesInsights.Samples.SampleLogger;
+
+namespace Azure.IoT.TimeSeriesInsights.Samples
+{
+    internal class InstancesSamples
+    {
+        /// <summary>
+        /// </summary>
+        public async Task RunSamplesAsync(TimeSeriesInsightsClient client)
+        {
+            PrintHeader("TIME SERIES INSIGHTS INSTANCES SAMPLE");
+
+            // Figure out how many keys make up the Time Series Id
+            TimeSeriesModelSettings modelSettings = await client.ModelSettings.GetAsync().ConfigureAwait(false);
+
+            TimeSeriesId instanceId = modelSettings.TimeSeriesIdProperties.Count switch
+            {
+                1 => new TimeSeriesId("key1"),
+                2 => new TimeSeriesId("key1", "key2"),
+                3 => new TimeSeriesId("key1", "key2", "key3"),
+                _ => throw new Exception($"Invalid number of Time Series Insights Id properties."),
+            };
+
+            #region Snippet:TimeSeriesInsightsSampleCreateInstance
+
+            // Create a Time Series Instance object with the default Time Series Insights type Id.
+            // The default type Id can be obtained programmatically by using the ModelSettings client.
+            var instance = new TimeSeriesInstance(instanceId, "1be09af9-f089-4d6b-9f0b-48018b5f7393")
+            {
+                Name = "instance1",
+            };
+
+            var tsiInstancesToCreate = new List<TimeSeriesInstance>
+            {
+                instance,
+            };
+
+            Response<TimeSeriesOperationError[]> createInstanceErrors = await client
+                .Instances
+                .CreateOrReplaceAsync(tsiInstancesToCreate)
+                .ConfigureAwait(false);
+
+            // The response of calling the API contains a list of error objects corresponding by position to the input parameter
+            // array in the request. If the error object is set to null, this means the operation was a success.
+            if (createInstanceErrors.Value[0] == null)
```

Comment:
wondering if we should do foreach. In this particular case we are just adding one instance but tomorrow if someone modifies, it's likely to miss this check.

Reply:
Good idea!

Reply:
Will change for the rest of them as well.